src/jit/lowerxarch.cpp

```diff
@@ -2296,6 +2296,34 @@ void Lowering::ContainCheckSIMD(GenTreeSIMD* simdNode)
 #endif // FEATURE_SIMD
 
 #ifdef FEATURE_HW_INTRINSICS
+//----------------------------------------------------------------------------------------------
+// IsContainableHWIntrinsicOp: Return true if 'node' is a containable HWIntrinsic op.
+//
+//  Arguments:
+//     node - The hardware intrinsic node.
+//
+bool Lowering::IsContainableHWIntrinsicOp(GenTree* node)
+{
+    if (!node->OperIsHWIntrinsic())
+    {
+        return false;
+    }
+
+    // TODO-XArch: Update this to be table driven, if possible.
+
+    NamedIntrinsic intrinsicID = node->AsHWIntrinsic()->gtHWIntrinsicId;
+
+    switch (intrinsicID)
+    {
+        case NI_SSE_LoadAlignedVector128:
+        case NI_SSE_LoadVector128:
```

Comment:
`LoadScalarVector128` should also be handled here, but I need to make sure it is emitted properly (since it is a `m32` instead of a `m128`)

Reply:
> but I need to make sure it is emitted properly (since it is a m32 instead of a m128)

Do you mean only folding `AddScalar(v1, LoadScalarVector128(address))` but not folding `AddScalar(v1, LoadVector128(address))`?

Reply:
Well specifically for `Add(v1, LoadScalar(address)) `, which would cause a read of too many bits, if folded.

I thin the other (`AddScalar(v1, LoadVector128(address)) `) is safe to fold (we should still determine if we want to do such an optimization) since the read isn't stored and the upper bits don't impact the final result

Reply:
We may not want to make the second optimization since the full read has potentially observable side effects (it could cause an AV exception, if you read past the end of an array for example or if it was an aligned read of an unlined address, etc).